crates/sparrow-compiler/src/data_context.rs

```diff
@@ -60,6 +61,10 @@ impl DataContext {
         self.table_info.get(&id)
     }
 
+    pub fn table_info_mut(&mut self, id: TableId) -> Option<&mut TableInfo> {
```

Comment:
Previously, DataContext was only used for compiling queries. Now, it's the easiest place to collect the data that we're going to run on (we don't have any way of plumbing "here is some data") today -- we just plumb "here are the files". This should change if we continue down this route.